compiler/hash-lower/src/lower_ty.rs

```diff
@@ -65,22 +66,21 @@ impl<'ir> BuilderCtx<'ir> {
     /// duplicate work.
     pub(crate) fn ty_id_from_tir_ty(&self, id: TyId) -> IrTyId {
         self.with_cache(id, || {
-            self.map_ty(id, |ty| {
-                // We compute the "uncached" type, and then it will be added to the
-                // cache if it is not already present. For data types, since they can
-                // be defined in a recursive way, the `ty_from_tir_data` will deal with
-                // its own caching, but we still want to add an entry here for `TyId` since
-                // we want to avoid computing the `ty_from_tir_data` as well.
-                let result = match ty {
-                    Ty::Data(data_ty) => self.ty_from_tir_data(*data_ty),
-
-                    // Hot path for unit types.
-                    Ty::Tuple(tuple) if tuple.data.is_empty() => COMMON_IR_TYS.unit,
-                    _ => self.uncached_ty_from_tir_ty(id, ty),
-                };
-
-                (result, false)
-            })
+            let ty = id.borrow();
```

Comment:
I meant to just do an `id.map(...)` here

Reply:
It's the same thing

Reply:
But surely the deref does the same thing as `value()` right?

Reply:
&* is a syntactic no op, it just casts from ref to &